### DISCUSSION TITLE
CPU temp / speed and Fan speed ?

### DISCUSSION THREAD

**Discussion by sibelle-labs** [dougiteixeira/proxmoxve#285] [Feature request] [answered]
Hi!
Maybe for better tracking of power consume, can you add please CPU temp / speed and Fan speed ?
Thanks a lot in advance,
cheers

**Comment by moshoari** [comment]
We need to follow up on this, as I can see that many people are asking for this upgrade.

**Reply by danielholm** [reply]
Indeed. I found another integration, which I haven't tried yet though. Is there API endpoints that could be used in vanilla proxmox?
https://github.com/mrkvka/ha-proxmox-cpu-ctl

**Comment by mZ738** [comment, marked as answer]
Hardware readings are in since 5.0.0: temperatures, fan speeds, voltages and power from `lm-sensors`, through [PVE-mods](https://github.com/Meliox/PVE-mods) on the node — the integration reads them from the node status the mod extends, no extra API calls. Both the current `node_info` package and the legacy script are recognised. How to set it up, which chips are classified how, and why readings come and go with the mod's on-demand collector: [Hardware sensors](https://github.com/dougiteixeira/proxmoxve/blob/main/docs/hardware-sensors.md). 5.1.0 ([release notes](https://github.com/dougiteixeira/proxmoxve/releases/tag/5.1.0)) is out today; I am closing this as done — please open a new thread if a chip of yours is not picked up.
